basket/news/tests/test_tasks.py

```diff
@@ -286,19 +310,173 @@ def test_credit_card_expiring(self, sfmc_mock, sfdc_mock, gud_mock):
             'dude@example.com', 'dude@example.com')
 
     def test_customer_created_customer_found(self, sfdc_mock, gud_mock):
-        pass
+        user_data = {
+            'first_name': 'Jeffrey',
+            'last_name': '_',
+        }
+
+        gud_mock.return_value = user_data
+
+        process_subhub_event_customer_created({
+            'name': 'Jeffrey Lebowski',
+            'email': 'thedude@thedude.io',
+            'user_id': '1234',
+            'customer_id': 'cus_1234',
+        })
+
+        sfdc_mock.update.assert_called_with(user_data, {
+            'fxa_id': '1234',
+            'payee_id': 'cus_1234',
+            'last_name': 'Lebowski',
+        })
 
     def test_customer_created_customer_not_found(self, sfdc_mock, gud_mock):
-        pass
+        gud_mock.return_value = None
+
+        process_subhub_event_customer_created({
+            'name': 'Walter Sobchak',
+            'email': 'water@thedude.io',
+            'user_id': '1234',
+            'customer_id': 'cus_1234',
+        })
+
+        sfdc_mock.update.assert_not_called()
+
+        sfdc_mock.add.assert_called_with({
+            'fxa_id': '1234',
+            'payee_id': 'cus_1234',
+            'first_name': 'Walter',
+            'last_name': 'Sobchak',
+            'email': 'water@thedude.io',
+        })
 
     def test_payment_failed(self, sfdc_mock, gud_mock):
-        pass
+        gud_mock.return_value = {'id': '1234'}
+
+        data = {
+            'customer_id': 'cus_1234',
+            'amount_due': '1000',
+            'created': '1566305505',
+            'invoice_number': 'abc123',
+            'invoice_id': 'inv_abc123',
+            'subscription_id': 'sub_123',
+            'charge_id': '8675309',
+            'nickname': 'bowling',
+            'currency': 'us',
+        }
+
+        process_subhub_event_payment_failed(data)
+
+        gud_mock.assert_called_with(payee_id=data['customer_id'], extra_fields=['id'])
+
+        sfdc_mock.opportunity.create.assert_called_with({
+            'Amount': cents_to_dollars(data['amount_due']),
+            'CloseDate': iso_format_unix_timestamp(data['created']),
```

Comment:
this is fine, but a better test would be to hard code the expected output of this function and the one above for Amount. This will make sure things stay right even if one of these functions change or we swap it out.